nixos/tests/installer.nix

```diff
@@ -681,18 +684,23 @@ in {
 
     createPartitions = ''
       machine.succeed(
-          "flock /dev/vda parted --script /dev/vda -- mklabel msdos"
-          + " mkpart primary linux-swap 1M 1024M"
-          + " mkpart primary 1024M -1s",
+          "flock /dev/vda parted --script /dev/vda -- mklabel gpt"
+          + " mkpart ESP fat32 1M 100M"  # /boot
+          + " set 1 boot on"
+          + " mkpart primary linux-swap 100M 1024M"
+          + " mkpart primary 1024M 100%", # /
           "udevadm settle",
-          "mkswap /dev/vda1 -L swap",
+          "mkswap /dev/vda2 -L swap",
           "swapon -L swap",
-          "zpool create rpool /dev/vda2",
+          "zpool create rpool /dev/vda3",
           "zfs create -o mountpoint=legacy rpool/root",
           "mount -t zfs rpool/root /mnt",
           "zfs create -o mountpoint=legacy rpool/root/usr",
           "mkdir /mnt/usr",
           "mount -t zfs rpool/root/usr /mnt/usr",
+          "mkfs.vfat -n BOOT /dev/vda1",
+          "mkdir -p /mnt/boot",
+          "mount LABEL=BOOT /mnt/boot",
```

Comment:
I'd mount esp to /boot/efi since it's the typical grub-efi configuration. bpool should still be mounted to /boot

Reply:
grub's zfs support won't be tested if esp is mounted to /boot

Reply:
I am not sure this is a goal for the project to test ZFS support of GRUB, I want to test UEFI Boot support of GRUB

Reply:
Anyone is free to add tests for ZFS support of GRUB, but each time, it will be unmaintained, we will remove it because I am certainly not maintaining this

Reply:
> Anyone is free to add tests for ZFS support of GRUB

This is exactly what i did in https://github.com/NixOS/nixpkgs/pull/262982

Reply:
"zfsroot" for me reads as "test NixOS install on a ZFS root filesystem", not "test NixOS install on a ZFS boot partition"
Whatever, we can rename this `on-zfs-rootfs` and have what you want as `-with-zfs-boot`

What you mentioned is `on-ext3-rootfs` FWIW.

Reply:
> What you mentioned is on-ext3-rootfs FWIW.

It doesn't matter what fs is used for root if bootloader is stored on fat32 fs

Reply:
Right, I just think we need to rename, and we should not make your test fixed as a channel blocker, only a simple ZFS root boot on ESP in UEFI.

If you want to support ZFS support of GRUB, please add yourself as maintainers of GRUB and ZFS. No offense but it's tiring to deal with both of them on those areas especially when have no one to turn to.
Otherwise, the test will be there on best effort basis.

Reply:
> > What you mentioned is on-ext3-rootfs FWIW.
> 
> It doesn't matter what fs is used for root if bootloader is stored on fat32 fs

Correct, but we are testing installer cycles, we want to know if the whole experience works out of the box.
Yes, we know that bootloading via ESP FAT32 will work.

Reply:
> Right, I just think we need to rename, and we should not make your test fixed as a channel blocker, only a simple ZFS root boot on ESP in UEFI.

I agree, I just think the existing test should remain with /boot on zfs pool, the name of the test doesn't really matter

> Otherwise, the test will be there on best effort basis

This is acceptable

Reply:
Gotcha, let's do the appropriate renames and send all of this